### PR TITLE
SparkSQL: Allow for any ordering of create table clauses

### PR DESCRIPTION
**Pull request by bmorck**
### Brief summary of the change made

This change updates `CREATE TABLE` statement clauses in sparksql to accept any ordering of the clauses and to accept comments.

A test has been added to `test/fixtures/dialects/sparksql/create_table_datasource.sql` to test this.

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_sparksql.py [hunk 1]
```diff
@@ -1121,11 +1121,15 @@ class CreateTableStatementSegment(ansi.CreateTableStatementSegment):
             optional=True,
         ),
         Sequence("USING", Ref("DataSourceFormatGrammar"), optional=True),
-        Ref("RowFormatClauseSegment", optional=True),
-        Ref("StoredAsGrammar", optional=True),
-        Ref("OptionsGrammar", optional=True),
-        Ref("PartitionSpecGrammar", optional=True),
-        Ref("BucketSpecGrammar", optional=True),
+        AnyNumberOf(
```
Comment (alanmcruickshank) [inline src/sqlfluff/dialects/dialect_sparksql.py:1124]: Rather than `AnyNumberOf(..., max_times_per_element=1)` you can just use `AnySetOf`.

Comment (alanmcruickshank) [reply src/sqlfluff/dialects/dialect_sparksql.py:1124]: In that case you could also make the `AnySetOf` grammar `optional`, and then you wouldn't need the `optional=True` on any of the inner segments.

File: src/sqlfluff/dialects/dialect_sparksql.py [hunk 2]
```diff
@@ -2756,6 +2760,22 @@ class PropertyNameSegment(BaseSegment):
...
+#         Ref("SingleIdentifierGrammar"),  # Column name
+#         Ref("DatatypeSegment"),  # Column type
+#         Bracketed(Anything(), optional=True),  # For types like VARCHAR(100)
+#         AnyNumberOf(
+#             Ref("ColumnConstraintSegment", optional=True),
+#         ),
+#         Ref("CommentGrammar", optional=True)
+#     )
+
+
```
Comment (alanmcruickshank) [inline src/sqlfluff/dialects/dialect_sparksql.py:2778]: Can you remove this new comment? I don't think it's necessary.

**Review by alanmcruickshank — CHANGES_REQUESTED** [review]
Looks like you've left an unnecessary comment in the dialect file. There's also a cleaner way of achieving what you're trying to do using `AnySetOf`.

File: src/sqlfluff/dialects/dialect_sparksql.py (hunk 1 above)
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_sparksql.py:1124]: AnySetOf may work better here?

File: src/sqlfluff/dialects/dialect_sparksql.py [hunk 3]
```diff
@@ -2756,6 +2760,22 @@ class PropertyNameSegment(BaseSegment):
     )
 
 
+# class ColumnDefinitionSegment(ansi.ColumnDefinitionSegment):
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_sparksql.py:2763]: If this code isn't used can we delete?

File: src/sqlfluff/dialects/dialect_sparksql.py (hunk 1 above)
Comment (alanmcruickshank) [reply src/sqlfluff/dialects/dialect_sparksql.py:1124]: snap :)

Comment (bmorck) [reply src/sqlfluff/dialects/dialect_sparksql.py]: Got it! Thanks for the heads up on `AnySetOf` -- wasn't aware of this.

**Review by WittierDinosaur — APPROVED** [review]
LGTM

**Pull request merged**